pkg/_internal/conditions/conditions.go

```diff
@@ -0,0 +1,52 @@
+package conditions
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/clock"
+)
+
+const (
+	ConditionTypeReady string = "Ready"
+)
+
+var Clock clock.Clock = clock.RealClock{}
+
+func HasCondition(existingConditions []metav1.Condition, c metav1.Condition) bool {
+	for _, cond := range existingConditions {
+		if c.Type == cond.Type && c.Status == cond.Status {
+			return true
+		}
+	}
+	return false
+}
+
+func SetCondition(conditions []metav1.Condition, generation int64, conditionType string, status metav1.ConditionStatus, reason, message string) []metav1.Condition {
```

Comment:
This is useful.
I'll work it in to the gateway reconcile pr around [here](https://github.com/Kuadrant/multi-cluster-traffic-controller/pull/84/files#diff-6afa7d73841a5c0ebdb78eaf859ce5a400203a037f871440940b5347c3a97376R256-R304)